src/main/java/com/yelp/nrtsearch/server/luceneserver/SuggestSearchHandler.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver;
+
+import com.yelp.nrtsearch.server.grpc.OneSuggestSearchResponse;
+import com.yelp.nrtsearch.server.grpc.SuggestSearchRequest;
+import com.yelp.nrtsearch.server.grpc.SuggestSearchResponse;
+import com.yelp.nrtsearch.server.luceneserver.field.ContextSuggestFieldDef;
+import com.yelp.nrtsearch.server.luceneserver.field.FieldDef;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.lucene.document.Document;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.index.IndexableField;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.suggest.document.CompletionQuery;
+import org.apache.lucene.search.suggest.document.ContextQuery;
+import org.apache.lucene.search.suggest.document.FuzzyCompletionQuery;
+import org.apache.lucene.search.suggest.document.PrefixCompletionQuery;
+import org.apache.lucene.search.suggest.document.SuggestIndexSearcher;
+import org.apache.lucene.search.suggest.document.TopSuggestDocs;
+
+public class SuggestSearchHandler implements Handler<SuggestSearchRequest, SuggestSearchResponse> {
+  @Override
+  public SuggestSearchResponse handle(
+      IndexState indexState, SuggestSearchRequest suggestLookupRequest) throws HandlerException {
+    ShardState shardState = indexState.getShard(0);
+    indexState.verifyStarted();
+    IndexReader indexReader;
+    try {
+      indexReader = shardState.acquire().searcher.getIndexReader();
```

Comment:
The `shardState.acquire()` needs to be balanced with a `shardState.release(<SearcherAndTaxonomy>)` when the operation finishes or fails. See the `SearchHandler` https://github.com/Yelp/nrtsearch/blob/master/src/main/java/com/yelp/nrtsearch/server/luceneserver/SearchHandler.java#L110

Reply:
Resolved it for now, but I need to revisit the exception handling and utilize a new `SuggestSearchHandlerException` for all exceptions in this class.